cpp/tests/strings/factories_test.cu

```diff
@@ -91,8 +91,14 @@ TEST_F(StringsFactoriesTest, CreateColumnFromPair)
 
   // check string data
   auto strings_data = cudf::strings::create_offsets(strings_view);
-  thrust::host_vector<char> h_chars_data(strings_data.first);
-  thrust::host_vector<cudf::size_type> h_offsets_data(strings_data.second);
+  thrust::host_vector<char> h_chars_data(strings_data.first.size());
+  thrust::host_vector<cudf::size_type> h_offsets_data(strings_data.second.size());
+  CUDA_TRY(cudaMemcpy(
+    h_chars_data.data(), strings_data.first.data(), h_chars_data.size(), cudaMemcpyDeviceToHost));
+  CUDA_TRY(cudaMemcpy(h_offsets_data.data(),
+                      strings_data.second.data(),
+                      h_offsets_data.size() * sizeof(cudf::size_type),
+                      cudaMemcpyDeviceToHost));
```

Comment:
Hmmm, identical code to the section you modified in column_utilities `to_host` repeated 3 times in this file. DRY this into a standalone utility?

I think 

```
template <typename T>
std::vector<T> to_host_async(rmm::device_uvector<T> const& v, rmm::cuda_stream_view stream);
```
should be added to vector_factories.hpp for use in libcudf and in tests. Or it could be called `make_vector_from_device_uvector_async` if we need it to sound like the factory it is...